### ISSUE TITLE
Cloning `Ffi_ArrowArray` causes segfault on drop.

### ISSUE THREAD

**Issue by zirconium-n** [jorgecarleitao/arrow2#880]
Following code easily segfault. `Ffi_ArrowArray`'s `Clone` or `Drop` is incorrect, or did I break contract of `export_array_to_c`?

``` rust 
use std::sync::Arc;

use arrow2::array::UInt32Array;
use arrow2::ffi::{Ffi_ArrowArray, export_array_to_c};

fn main() {
    let arr = Arc::new(UInt32Array::from_vec(vec![1, 2, 3, 4]));
    let mut arr_ffi = Ffi_ArrowArray::empty();

    unsafe {
        export_array_to_c(arr.clone(), &mut arr_ffi as *mut _);
    };

    let _ = arr_ffi.clone();

    // Segfault on drop
}
```

**Comment by zirconium-n** [comment]
Yes, I agree removing `#[derive(Clone)]` is the best solution.

Another way would be removing `impl Drop` and provide a `unsafe fn drop_array`, but it's probably too hard to use it correctly.

P.S. `ArrowSchema` and `ArrowArrayStream` have the same problem too.

**Comment by jorgecarleitao** [comment]
I have now fixed this in main, v0.7, v0.8 and v0.9. I has created an advisory, RUSTSEC-2022-0012 via https://github.com/rustsec/advisory-db/pull/1204

Thank you again for this report that positively contributes to the soundness of this crate in particular and rust's ecosystem more broadly.
